### PR TITLE
Added support for the "replace" database command

### PR DESCRIPTION
**Pull request by annando**
`replace` is a standardized way of inserting and handle key duplicates. It isn't used at the moment. In the future it should be used whenever tables values are inserted and there is an unique key.

### CODE REVIEW

**Comment by MrPetovan** [comment]
I though the `REPLACE` keyword was specific to MySQL?

File: src/Database/DBA.php [hunk 1]
```diff
@@ -292,6 +292,20 @@ public static function insert($table, $param, $on_duplicate_update = false)
 		return DI::dba()->insert($table, $param, $on_duplicate_update);
 	}
 
+	/**
+	 * Replace a row of a table
```
Comment (MrPetovan) [inline src/Database/DBA.php:296]: Please be more specific, something like
```suggestion
	 * Inserts a row with the provided data in the provided table.
	 * If the data corresponds to an existing row through a UNIQUE or PRIMARY index constraints, it updates the row instead.
```

File: src/Database/Database.php [hunk 2]
```diff
@@ -1006,6 +1006,33 @@ public function insert($table, array $param, bool $on_duplicate_update = false)
 		return $this->e($sql, $param);
 	}
 
+	/**
+	 * Replace a row of a table
```
Comment (MrPetovan) [inline src/Database/Database.php:1010]: Please be more specific, something like
```suggestion
	 * Inserts a row with the provided data in the provided table.
	 * If the data corresponds to an existing row through a UNIQUE or PRIMARY index constraints, it updates the row instead.
```

**Comment by annando** [comment]
@MrPetovan yeah, you seem to be right. I heard that more SQL servers would support this command. But a quick research hadn't shown any evidence.

**Comment by annando** [comment]
I haven't found that command :-)

**Comment by MrPetovan** [comment]
> @MrPetovan yeah, you seem to be right. I heard that more SQL servers would support this command. But a quick research hadn't shown any evidence.

We just have to remember this we we implement PostgreSQL support by using `ON DUPLICATE KEY UPDATE` instead.

**Comment by annando** [comment]
It seems to be `INSERT INTO ... ON CONFLICT ...`
https://www.postgresql.org/docs/current/sql-insert.html

**Comment by MrPetovan** [comment]
Fair enough, I thought `ON DUPLICATE KEY` was standard SQL.

**Pull request merged**